graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/SocketModuleBuiltins.java

```diff
@@ -113,4 +210,79 @@ String doGeneric() {
             }
         }
     }
+
+    @Builtin(name = "getaddrinfo", parameterNames = {"host", "port", "family", "type", "proto", "flags"})
+    @GenerateNodeFactory
+    public abstract static class GetAddrInfoNode extends PythonBuiltinNode {
+        @Specialization
+        @TruffleBoundary
+        Object getAddrInfo(String host, int port, int family, int type, int proto, PInt flags) {
+            return getAddrInfo(host, port, family, type, proto, flags.asInt());
+        }
+        @Specialization
+        @TruffleBoundary
+        Object getAddrInfo(String host, String port, int family, int type, int proto, PInt flags) {
+            return getAddrInfo(host, port, family, type, proto, flags.asInt());
+        }
+        @Specialization
+        @TruffleBoundary
+        Object getAddrInfo(String host, int port, int family, int type, int proto, int flags) {
+            InetAddress[] adresses = resolveHost(host);
+            List<Service> serviceList = new ArrayList();
```

Comment:
It looks like you're not modifying the serviceList in `mergeAddressesAndServices`. Can you instead use
```
Service[] serviceList = new ServiceList[2];
serviceList[0] = new Service(port, "tcp");
serviceList[1] = new Service(port, "udp");
```
?

Reply:
If you can, the TruffleBoundary here can also be removed

Reply:
> It looks like you're not modifying the serviceList in `mergeAddressesAndServices`. Can you instead use
> 
> ```
> Service[] serviceList = new ServiceList[2];
> serviceList[0] = new Service(port, "tcp");
> serviceList[1] = new Service(port, "udp");
> ```
> 
> ?

Problem is that parseServices needs the ArrayLists add, so to change this one, we would need to convert those lists in the services Map too.